### ISSUE TITLE
set_and_check broken for scientific notation floats

### ISSUE THREAD

**Issue by wcpettus** [project8/dragonfly#66]
The check functionality is not properly parsing scientific notation floats.  Example that breaks:

```
  - name: rsa_default_config
    module: MultiDo
    targets:
      - target: rsa_central_frequency
        payload_field: value_raw
        units: Hz
        default_set: 1140E+6
```

The output said it was comparing 1140E+6 with 1140000000.0 and found them not equal, so something is not being cast correctly.  A similar (or identical?) problem was noticed in the run_scripting.

**Comment by laroque** [comment]
If you have a line number from where this is happening (whatever print is telling you the comparison), you can add a line `import ipdb;ipdb.set_trace()` to make a debugger break point. From there you should be able to try inspecting the types and values and test whatever casting is being done. From a shell `1140000000.0 == 1140E+6` returns True so if there a casting problem it is likely not the automatic casting.

That said, we should generally define a finite tolerance for these sorts of checks since instruments may use a different precision than their input and we wouldn't want to get a False just because we have different floating point precision.

**Comment by wcpettus** [comment]
fixed in release v1.2.4
